logstash-core/src/main/java/org/logstash/instrument/metrics/FlowMetric.java

```diff
@@ -34,8 +46,13 @@ public FlowMetric(final String name,
     }
 
     public void capture() {
-        final Capture previousHead = head.getAndSet(doCapture());
-        instant.set(previousHead);
+        final Capture newestHead = doCapture();
+        final Capture previousHead = head.getAndSet(newestHead);
+        // rotate our instant forward only
+        instant.getAndAccumulate(previousHead, (current, given) -> {
```

Comment:
This is too naive and if our capture is called too frequently without a break can cause the "current" capture window between our head and this previous capture to grow indefinitely.

I was trying too hard to defer the work for our rolling averages that is very similar to the work I already spiked a few months ago, but need to pull in at least a little of that complexity to make sure our "current" window isn't too small to be meaningful.

Reply:
Resolved, by reducing the over-capture frequency guard to 100ms, and adding in-line docs. 

I consider this "good enough" for three combined reasons:
 - we are aiming to capture at a 5s frequency for the technology preview, so the likelihood of accidentally relentlessly capturing on a less-than-100-ms frequency is next to zero
 - we plan to _generically_ implement rolling averages in a near-future release, which will support something like a `10s@1s` definition for our "current" rate (read: a 10-second window, maintaining just enough captures for a 1-second granularity)
 - even _if_ we hit this edge-case, the net effect is that our "current" flow metric merely accurately reports the rate over a bigger time-frame than hoped.